File: public/src/bazzbomber.c
#include "bazzbomber.h"
#include "maths.h"
#include "sprite_eng.h"
#include "types.h"

void BAZZBOMBER_constructor(Bazzbomber *this, Sprite *sprite, fix32 posX, fix32 posY) {
    
        this->sprite = sprite;
        this->posX = posX;
        this->posY = posY;
    }

void BAZZBOMBER_update(Bazzbomber *this) {
        this->posX -= FIX32(1);
    }

File: src/app/bazzbomber.ts
import { FIX32 } from "./maths";
import { Sprite } from "./sprite_eng";
import { fix32 } from "./types";

export class BazzBomber {
    posX: fix32;
    posY: fix32;
    sprite: Sprite;
    hFlip: boolean;

    constructor(sprite: Sprite, posX: fix32, posY: fix32) {
        this.sprite = sprite;
        this.posX = posX;
        this.posY = posY;
    }

    update() {
        this.posX -= FIX32(1);
    }
}